src/util/metrics/mod.rs

```diff
@@ -60,14 +60,14 @@ pub fn run_prometheus(
     Some(handler)
 }
 
-pub fn dump() -> String {
+pub fn dump() -> Result<String, Error> {
     let mut buffer = vec![];
     let encoder = TextEncoder::new();
     let metric_familys = prometheus::gather();
     for mf in metric_familys {
         if let Err(e) = encoder.encode(&[mf], &mut buffer) {
-            warn!("ignore prometheus encoding error: {:?}", e);
+            return Err(e);
```

Comment:
Warn the error?

Reply:
IMO, I prefer ignoring the error and not return it. We still want to fetch other metrics if possible.